src/Identifier/OrmIdentifier.php

```diff
@@ -96,17 +98,27 @@ protected function _findUser($username, $password = null)
     /**
      * Get query object for fetching user from database.
      *
-     * @param string $username The username/identifier.
+     * @param string $identifier The username/identifier.
      * @return \Cake\ORM\Query
      */
-    protected function _query($username)
+    protected function _query($identifier)
     {
         $config = $this->_config;
         $table = TableRegistry::get($config['userModel']);
 
-        $options = [
-            'conditions' => [$table->aliasField($config['fields']['username']) => $username]
-        ];
+        if (is_array($config['fields']['username'])) {
+            $conditions = [];
+            foreach ($config['fields']['username'] as $field) {
+                $conditions[$table->aliasField($field)] = $identifier;
+            }
+            $conditions = [
+                'OR' => $conditions
+            ];
+        } else {
+            $conditions = [$table->aliasField($config['fields']['username']) => $identifier];
+        }
+
+        $options = compact('conditions');
```

Comment:
Can't we put this whole piece of added code into a new method? Something like `buildOptions()`? Makes it more easy to override the method. And will probably make the code rating more happy and going up.